nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/template/TemplateInfoHolder.java

```diff
@@ -67,29 +67,22 @@ public String getLocation() {
      * @return Parsed Object.
      * @throws Exception If any error occurs in parsing.
      */
-    @SuppressWarnings("rawtypes")
-    private TemplateInfo parse(InputStream configStream) throws Exception {
+    private static TemplateInfo parse(InputStream configStream) throws Exception {
         if (configStream == null) {
-            throw new DomainException("Invalid stream");
-        }
-        try {
-            URL schemaUrl = getClass().getClassLoader().getResource(TEMPLATE_INFO_SCHEMA_PATH);
-            JAXBContext context = JAXBContext.newInstance(TemplateInfo.class.getPackage().getName());
-            Unmarshaller unmarshaller = context.createUnmarshaller();
-            SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
-            Schema schema = schemaFactory.newSchema(schemaUrl);
-            unmarshaller.setSchema(schema);
-            InputSource is = new InputSource(configStream);
-            SAXSource source = new SAXSource(is);
-            Object obj = unmarshaller.unmarshal(source);
-            return obj instanceof JAXBElement ? (TemplateInfo) (((JAXBElement) obj).getValue()) : (TemplateInfo) obj;
-        } finally {
-            try {
-                configStream.close();
-                configStream = null;
-            } catch (IOException e) {
-                /** Ignore */
-            }
+            throw new DomainException("Invalid stream: null");
         }
+        URL schemaUrl = TemplateInfoHolder.class.getClassLoader().getResource(TEMPLATE_INFO_SCHEMA_PATH);
+        JAXBContext context = JAXBContext.newInstance(TemplateInfo.class.getPackage().getName());
+        Unmarshaller unmarshaller = context.createUnmarshaller();
+        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        schemaFactory.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
```

Comment:
External entities not allowed ony here. Is it correct?

Reply:
I remember that also disabling some protocols helps, and GlassFish here should have all dependencies locally available. I remember some old versions were missing something and we discovered it on production which did not have access to the internet. This should cause that some tests would fail.
External entities are disabled by the "secure processing feature".
I think SpotBugs will report also other places, but I am going per module.
I will add more references.

Reply:
Hmm, however it seems that in this case we can disable both. I will recheck the second change.